common/http/platform/beast/http.cpp

```diff
@@ -607,28 +610,29 @@ void Client::ReadHeaderHandler(const error_code &ec, size_t num_read) {
 	}
 
 	auto cancelled = cancelled_;
+	status_ = TransactionStatus::HeaderHandlerCalled;
 	CallHandler(header_handler_);
 	if (*cancelled) {
 		return;
 	}
 
-	if (body_status_ == BodyReadingStatus::None) {
+	if (status_ == TransactionStatus::HeaderHandlerCalled) {
```

Comment:
At this point we have called the user header handler, where we expect that (the user) calls `MakeBodyAsyncReader` where `status_` is set to `ReaderCreated`. So which is the path when there is no body (when `response_body_length_` is 0)? Wouldn't then we always hit this `if` condition?

Reply:
`response_body_length_` cannot be zero, because we check `response_body_read_ >= response_body_length_` some lines above this, and since the former is initialized to zero, we will return from there in that case.

Reply:
I can add a comment about it.

Reply:
Now I see it. Thank you.